database/handler/process_handler.py

```diff
@@ -223,7 +231,18 @@ def update_process_reference(input_dict: dict) -> dict:
     :type input_dict: dict
     :return: Status dict
     """
-    return core.success_handler()
+
+    process = Process.nodes.get(uid=input_dict['uid'])
+
+    component_list = process.hasComponent.all()
+
+    for component in component_list:
+        rel = process.hasComponent.relationship(component)
+        if rel.weight == input_dict['old_weight']:
```

Comment:
Brauchen wir diese Überprüfung? Reicht es nicht aus, das "new_weight" direkt zu setzen?

Reply:
Nein, das reicht nicht aus. In diesem Fall ist das weight unser Unique Identifier um die richtige Relation zu erkennen. Ansonsten würden wir einfach alle Relations ändern. Da wir aber nur die entsprechende Relation anpassen wollen, müssen wir diese irgendwie erkennen und das funktioniert hier über weight.